crates/sui-framework/sources/dynamic_field.move

```diff
@@ -56,6 +56,21 @@ public fun add<Name: copy + drop + store, Value: store>(
     add_child_object(object_addr, field)
 }
 
+spec add {
+    pragma opaque;
+    aborts_if [abstract] sui::prover::uid_has_field(object.id.bytes, name);
+    modifies [abstract] global<object::DynamicFields<Name>>(object.id.bytes);
```

Comment:
Nit: is it possible to use a function instead of directly accessing the field `object.id.bytes` here and below?

Reply:
I finessed it as much as I could, at the cost of a bit longer expressions, but probably still worth it.